added instructions for building aginst modded packages, as per #5066

telamonian opened the pull request:
I recently made some changes to phosphor and wanted to test them out against JupyterLab. I worked out a fast/simple way to do it, so I want to add a how-to to the docs in case anyone else runs into the same issue.

blink1073 commented on CONTRIBUTING.md:
```diff
@@ -327,6 +327,27 @@ a package by importing from it in the TypeScript file, and then running:
+2. Reinstall original version of the external package in JupyterLab
+    - run `jlpm install --check-files`
+
+You can then (re)build JupyterLab and everything should be back to default.
```
Thanks! Do you mind added the note about other libraries sometimes needing to be linked because of shared state (like the message loop)?

telamonian replied:
Of course

telamonian commented:
I added a note about the shared state pitfall.

Also, it looks like `prettier` ran and changed the formatting of `CONTRIBUTING.md` as part of the first commit, but somehow the prettied version wasn't included in that first commit. Maybe there's a small bug in the commit hook?

blink1073 commented:
Hmm, this might be a bug in the precommit hook from `husky`, looks like husky 1.0 is coming [soon](https://github.com/typicode/husky).

telamonian commented on CONTRIBUTING.md:
```diff
@@ -327,6 +327,35 @@ a package by importing from it in the TypeScript file, and then running:
+
+### Linking/Unlinking Packages to JupyterLab
+
+If you want to make changes to one of JupyterLab's external packages (for example, [phosphor](https://github.com/phosphorjs/phosphor)) and test them out against your copy of JupyterLab, you can easily do so using the `link` command:
```
Done

blink1073 commented:
Thanks!
